components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -114,12 +117,14 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
-	err = updateWorkspaceStatus(ctx, &workspace, workspacePods, r.Config)
+	oldStatus := workspace.Status
```

Comment:
```suggestion
	oldStatus := workspace.Status.DeepCopy()
```
to ensure e.g. pointers are also copied, don't think it makes any difference now but might in the future?